### PR TITLE
ORV2-2870 - Regression: Editing year on vehicle will frequently save a random year when saving

### PR DESCRIPTION
**Pull request by glen-aot**
# Description

create disableMouseWheelInputOnNumberField function and pass to all applicable fields with inputType: 'number'

Fixes [#ORV2-2870](https://moti-imb.atlassian.net/browse/ORV2-2870?atlOrigin=eyJpIjoiN2ExMTBlNzEyY2QxNDZkOThlMTJlZGM1MTljZTUyOTIiLCJwIjoiaiJ9)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1640-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1640-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1640-dops.apps.silver.devops.gov.bc.ca/api)
- [Policy](https://onroutebc-1640-policy.apps.silver.devops.gov.bc.ca/api)
- [Scheduler](https://onroutebc-1640-scheduler.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/common/helpers/disableMouseWheelInputOnNumberField.ts
Comment (zgong-gov) [inline frontend/src/common/helpers/disableMouseWheelInputOnNumberField.ts]: Can this same effect be achieved in another way? Because as is, this method creates a side effect which mutates the existing event, and also we might need to remove the event listener later when the page/component lifecycle ends? I'm wondering if there's another way to override the mouse wheel behaviour (maybe by passing in a prop somewhere if that's possible) without introducing side effects...

File: frontend/src/features/permits/pages/Application/components/form/VehicleDetails/VehicleDetails.tsx
Comment (zgong-gov) [inline frontend/src/features/permits/pages/Application/components/form/VehicleDetails/VehicleDetails.tsx]: I have made a lot of changes in the VehicleDetails component, which included removing some of these variables... You might need to wait until that PR is merged, and then we can consolidate your changes along with my changes.

Comment (glen-aot) [reply frontend/src/features/permits/pages/Application/components/form/VehicleDetails/VehicleDetails.tsx]: I didn't actually touch this code, I think my prettier must have formatted this file differently on save hence the changes here. Please disregard

File: frontend/src/common/helpers/disableMouseWheelInputOnNumberField.ts
Comment (glen-aot) [reply frontend/src/common/helpers/disableMouseWheelInputOnNumberField.ts]: The simplest way would be to stop using inputType: "number" in favor of a standard input, like we are using for unitNumber for example. However if we are set on using the number field then this method of attaching the event listener to the DOM element is the only way I can solve this problem, since React doesn't offer direct control over this.

My original attempt to solve this issue was something like this:

`
export const disableMouseWheelInput = (
  event: React.WheelEvent<HTMLInputElement>
) => {
  event.preventDefault(); 
};

<CustomFormComponent onWheel={disableMouseWheelInput} />
`

However, this will not work as indicated by the following error in the browser console:

"Unable to preventDefault inside passive event listener invocation"

I do agree however that we should remove the event listener, which I have added in my latest commit

**Review by zgong-gov — APPROVED** [review]

**Pull request merged**
